### ISSUE TITLE
[Bug] PropertyPane validations are not working as expected

### ISSUE THREAD

**Issue by aswathkk** [appsmithorg/appsmith#7485]
## Description

I tried to pass a random non-URL value to Video Widget's URL and it is not showing any error.
In the code, I could see that we are validating this field using Regex.

### Steps to reproduce the behaviour:

Add a video widget and pass in any non-URL to its URL field.

https://user-images.githubusercontent.com/10436935/133488535-225af348-359b-4ae4-96a7-fa67eba8b407.mov

### Important Details

- Version: [Cloud]
- OS: [MacOS]
- Browser [chrome]
- Environment [production]

**Comment by somangshu** [comment]
@aswathkk can you add note on what changes are required here to fix this validation issue here, This is a good issue to solve for a first time contributor

**Comment by errorinc0de** [comment]
Can I contribute to this issue?

**Comment by aswathkk** [comment]
Thank you @errorinc0de for showing interest. I've assigned it to you now.
Please take some time to go through the [Contributing Guidelines](https://github.com/appsmithorg/appsmith/blob/master/CONTRIBUTING.md).

You could see the configuration for VideoWidget's property pane [here](https://github.com/appsmithorg/appsmith/blob/0619e3798f3edff10de450a0dc54afde84220352/app/client/src/widgets/VideoWidget/widget/index.tsx#L37).
The validations are evaluated on `validate` function on [validations.ts](https://github.com/appsmithorg/appsmith/blob/0619e3798f3edff10de450a0dc54afde84220352/app/client/src/workers/validations.ts#L212) file.

In order to fix this issue, Take a look at how the Regex that we pass from VideoWidget's property pane is being evaluated [here](https://github.com/appsmithorg/appsmith/blob/0619e3798f3edff10de450a0dc54afde84220352/app/client/src/workers/validations.ts#L354).

**Comment by errorinc0de** [comment]
Thanks for assigning me the issue, I will work and try to resolve it as soon as possible and also thanks for sharing the files concerned with the bug.

@aswathkk

**Comment by Kathuria** [comment]
@aswathkk : Got it.

If I use npm for running project and packages - its failing and works with yarn. Although both does the same work, so I used npm even doc says yarn to use.

I reverted my TS lint fixes as well from my previous commit that gives error with yarn start. Added fix for this issue and this time it works as well. For ref: 
<img width="848" alt="Screenshot 2021-10-30 at 7 03 37 PM" src="https://user-images.githubusercontent.com/19428377/139534871-bf7923e0-9254-4356-a3c7-12fccc88c2f7.png">


Fix: RegExp was not used on regex and regex was compared with string, so failing that returns always true of url.
